src/others/components/FilterDropdown/FilterDropdown.tsx

```diff
@@ -92,44 +106,59 @@ export const FilterDropdown: FunctionComponent<FilterDropdownProps> = ({
         </Typography>
         {!singleValueFilter && <Chip size="small" color="primary" label={selectedFilterItemCount} />}
       </Button>
+
       {isFilterOpen && (
-        <ClickAwayListener onClickAway={toggleFilterList}>
-          <List
-            sx={{
-              borderRadius: "24px",
-              backgroundColor: "#fff",
-              marginTop: "16px",
-              maxHeight: "500px",
-              overflow: "auto",
-              position: "absolute",
-              zIndex: 1000,
-              minWidth: "300px",
-              color: "#000",
-            }}
-          >
-            {!singleValueFilter && (
-              <ListItem sx={{ justifyContent: "center" }}>
-                <Button color="primary" onClick={clearAllFilters}>
-                  {t("clear")}
-                </Button>
-              </ListItem>
-            )}
-            <ListItem>
-              <ListItemButton divider={true} onClick={clearAllFilters}>
-                <ListItemText color="#000">{t("all")}</ListItemText>
-                {checkboxListItemIcon(allFiltersSelected)}
-              </ListItemButton>
-            </ListItem>
-            {filterItems.map(({ id, text, selected }) => (
-              <ListItem key={id}>
-                <ListItemButton onClick={clickHandler(id, selected)}>
-                  <ListItemText>{text}</ListItemText>
-                  {checkboxListItemIcon(selected)}
-                </ListItemButton>
-              </ListItem>
-            ))}
-          </List>
-        </ClickAwayListener>
+        <>
+          {filterName === "Cities" ? (
```

Comment:
**issue**

I don't think this component should know about "Cities". How about passing the prop `hasSearch` instead?

Reply:
**question**

Shouldn't it be wrapper in `ClickAwayListener` as well?